### DISCUSSION TITLE
How to use system installed libraries?

### DISCUSSION THREAD

**Discussion by hector-vido** [keplerproject/luarocks#1405] [Q&A]
Hi guys!

I'm trying to build a .rockspec file that needs luasql-mysql:

```lua
package = "app"
version = "0.1-1"
source = {
   url = ""
}
description = {
   homepage = "",
   license = ""
}
dependencies = {
	'inspect',
	'lapis',
	'luasql-mysql'
}
build = {
   type = "builtin",
   modules = {
      app = "app.lua",
      config = "config.lua"
   }
}
```

Everything works if I install the headers:

```bash
apk add mariadb-dev
luarocks-5.1 build app-0.1-1.rockspec MYSQL_INCDIR=/usr/include/mysql
```

> Yes, https://luarocks.org/luasql-mysql-2.6.0-1.rockspec has a problem with unauthenticated git, I sent a merge request.

But now I don't want to compile this MySQL library, I want to use the one that I installed through package manager:

```bash
apk add lua-sql-mysql
```

The same thing happens with the **inspect** library, or any other.

There is a way to tell to luarocks that I want to use a local library?

**Comment by hishamhm** [comment]
> Yes, https://luarocks.org/luasql-mysql-2.6.0-1.rockspec has a problem with unauthenticated git, I sent a merge request.

If you upgrade to LuaRocks 3.8.0, it addresses the github authentication changes transparently!

> But now I don't want to compile this MySQL library, I want to use the one that I installed through package manager:

That will work seamlessly if you distro included the LuaRocks metadata in the package; otherwise, LuaRocks has no way to know that the package exists.

**Comment by hector-vido** [comment]
Hi @hishamhm thanks for the reply and the tip about version 3.8!

I though LuaRocks will look for C libraries without the need of these metadata from cpath.
So, LuaRocks consider only packages installed by itself or manually, but only because these metadata?

Thank you!
